tests/unit-tests/input/evdev/test_libinput_device.cpp

```diff
@@ -544,6 +545,46 @@ TEST_F(LibInputDeviceOnTouchScreen, process_event_handles_touch_down_events)
     process_events(touch_screen);
 }
 
+TEST_F(LibInputDeviceOnTouchScreen, process_event_ignores_uncorrelated_touch_up_events)
+{
+    MirTouchId slot = 3;
+
+    InSequence seq;
+
+    EXPECT_CALL(mock_builder, touch_event(_, _)).Times(0);
+    EXPECT_CALL(mock_sink, handle_input(_)).Times(0);
+
+    touch_screen.start(&mock_sink, &mock_builder);
+    env.mock_libinput.setup_touch_up_event(fake_device, event_time_1, slot);
+    env.mock_libinput.setup_touch_frame(fake_device, event_time_1);
+    process_events(touch_screen);
+}
+
+TEST_F(LibInputDeviceOnTouchScreen, process_event_ignores_excessive_touch_events_on_coordinate_0_0)
```

Comment:
I would like to see one more touch event to be added in the same frame, not on (0, 0), to verify that the whole frame is dropped.

Also another test where you have two touch events, one on (0, 0) and another not, to verify that the frame is not dropped.